### ISSUE TITLE
Problem with making geometry for networks loaded from matlab

### ISSUE THREAD

**Issue by arashmh** [PMEAL/OpenPNM-Examples#24]
Hi and cheers for your nice repo.
I'm trying to do some calculations on the network I have extracted by my own code using matlab. However, when I try to set the geometry like in your example:
`geom_name = pn.geometries()[0]`
`geom = pn.geometries(geom_nam)`
It makes a list and when I want to assign it to a physics object it raises the error that list object doesnt have a name, or when I want to assign this:
`geom['pore.diameter'] = 2*geom['pore.radius']`
It says : `TypeError: list indices must be integers or slices, not str`
Any ideas how to solve it?

**Comment by jgostick** [comment]
Hi @arashmh I'm hope you're finding it all useful.  As for your problem, you have actually stumbled upon one the problematic parts of the code that we will be fixing in V2.0 (hopefully sometime in 2017).  

I actually can't quite tell from your above code what the problem is...but let me try to explain it a bit:

* All objects in OpenPNM are `dictionaries`, so they are objects with various methods and attributes.  One of the attributes is the `name`` as in `pn.name` and `geom.name`.  
* Another attribute is a list of all geometries that have been associated with it as in `pn.geometries()`.  This method returns a `list` with all the geometries in it.  You access them with the `[0]` indexing.  So `pn.geometries()[0]` accesses a geometry object, which has a `name` attribute.  
* Basically, just add a `.name` to the end of your first line and see if that helps.

**Comment by arashmh** [comment]
Hi @jgostick and thanks for the code. Yes it looks very interesting , however I haven't had any success working with .mat file with openpm. I am trying to run at least one simulation on the sample .mat file you have provided in the repo and then move on to my own .mat file.

I followed the instructions [here](https://github.com/PMEAL/OpenPNM-Examples/blob/master/IO_and_Visualization/mat_load.md), and printed the type of objects to make sure about them and made a physics object based on that: 
`import OpenPNM as op`
`import scipy as sp`
`fname = 'example_network.mat'`
`pn = op.Network.MatFile(filename=fname, xtra_pore_data='type', xtra_throat_data='type')`
`op.export_data(network=pn, filename='matlab_test')`
`geom_name = pn.geometries()[0]`
`print('geom_name type: ',type(geom_name))`
`geom = pn.geometries(geom_name)`
`print('geom type: ',type(geom))`
`phys = op.Physics.GenericPhysics(network=pn, phase=water, geometry=geom)`

and I got this:
`geom_name type:  <class 'str'>`
`geom type:  <class 'list'>`
`Traceback (most recent call last):
  File "/Users/Arash/Documents/PhDformal/My papers/Dissolution/openpnm/test_mat.py", line 24, in <module>
    phys = op.Physics.GenericPhysics(network=pn, phase=water, geometry=geom)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/OpenPNM/Physics/__GenericPhysics__.py", line 66, in __init__
    pores = self._net.toindices(self._net['pore.' + geometry.name])
AttributeError: 'list' object has no attribute 'name'`

So geom is a list , and how can I add a .name to it ? `geom.name=geom_name` ? That raises the same error.
One more issue and that is `pn.geomtries()` sometimes has the `['boundary', 'GenericGeometry_gKjEv']` structure , while sometimes it swaps their places (boundary and geometry object) and therefore `pn.geometries()[0]` won't always give the geometry object name.

**Comment by arashmh** [comment]
I can also convert my .mat data into CSV format. Does the CSV class in IO  module work ? and if so , do you have a sample .csv file that I can use as a template?

Cheers

**Comment by jgostick** [comment]
Wow, I"m sorry...my original advice was WRONG!  Like I said, this part of the code is very old and rough, and we want to redo it in V2...but we're stuck with it for now.  

The actual problem with the above code is a missing `[0]` on `geom = pn.geometries(geom_name)[0]`.  I will fix the example asap.

**Comment by arashmh** [comment]
Yeah so I assume that importing .mat is a no go for now. How about the CSV import ? Does that work ? and do u have any template for that ?

**Comment by jgostick** [comment]
I would say yes, ditch the matfile, and point you toward a [new example I started](https://github.com/PMEAL/OpenPNM-Examples/pull/26/files#diff-4614b795a6107e825d3659f1ec0cbb67) this morning explaining the CSV format.  The example is not passing the unit tests yet, but I think it's just formatting...so it should be useful for you.

**Comment by jgostick** [comment]
I added one to the [fixtures directory](https://github.com/PMEAL/OpenPNM-Examples/tree/fix_IO_examples/fixtures) in the PR.
